File: shopapp/shopapp/src/main/java/com/project/shopapp/models/CartDetail.java
package com.project.shopapp.models;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "cart_details")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Builder
public class CartDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "cart_id", nullable = false) // Khóa ngoại đến Cart
    private Cart cart;
    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false) // Khóa ngoại đến Product
    private Product product;
    @Column(nullable = false)
    private Integer quantity = 1;
    @Column(nullable = false, precision = 15, scale = 2)
    private BigDecimal unitPrice;
    // totalPrice được tính tự động, không cần set giá trị
    @Column(precision = 15, scale = 2)
    private BigDecimal totalPrice;
}

File: shopapp/shopapp/shopapp_angular/src/app/models/cart-detail.ts
export class CartDetail {
  cart_detail_id: number;
  quantity: number;
  product_id: number;
  product_name: string;
  product_image: string;
  product_quantity: number;
  unit_price: number;
  total_price: number;

  constructor(
    cart_detail_id: number,
    product_id: number,
    quantity: number,
    product_name: string,
    product_image: string,
    product_quantity: number,
    unit_price: number,
    total_price: number
  ) {
    this.cart_detail_id = cart_detail_id;
    this.quantity = quantity;
    this.product_id = product_id;
    this.product_name = product_name;
    this.product_image = product_image;
    this.product_quantity = product_quantity;
    this.unit_price = unit_price;
    this.total_price = total_price;
  }
}
